### ISSUE TITLE
Not(NoneOf(a,b)) and Not(OneOf(a,b)) don't work properly

### ISSUE THREAD

**Issue by wvervuurt** [Respect/Validation#322]
Putting NoneOf or OneOf into a Not validator doesn't seem to work as it should.

Case: `$v = v::noneOf(v::numeric(), v::int())` and `$notV = v::not($v)`.
- OK: `$v->validate(13.37)` returns `false`, because 13.37 is numeric
- Problem: `$notV->validate(13.37)` returns `false` as well, even though 13.37 is numeric

Case: `$v = v::oneOf(v::numeric(), v::int())` and `$notV = v::not($v)`.
- OK: `$v->validate(13.37)` returns `true`, because 13.37 is numeric
- Problem: `$notV->validate(13.37)` returns `true` as well, even though 13.37 is numeric

Runnable code:

```php
$v = v::NoneOf(v::numeric(), v::int());
$notV = v::Not($v);
echo ($v->validate(13.37) ? 'true' : 'false');
echo " ";
echo ($notV->validate(13.37) ? 'true' : 'false');

echo '---';

$v = v::OneOf(v::numeric(), v::int());
$notV = v::Not($v);
echo ($v->validate(13.37) ? 'true' : 'false');
echo " ";
echo ($notV->validate(13.37) ? 'true' : 'false');
```

This outputs: `false false---true true`
What I expect is: `false true---true false`

It seems to me that a validator and its negation returning the same value is a bug. Am I missing something or doing something wrong here?

**Comment by maxw3st** [comment]
for the first pair I would expect to see: false true
you define v as being neither numeric nor an integer then have it validate a numeric value (false) followed by its negation, notV which is defined as being either numeric or an integer

for the second pair v is either numeric or an integer, and notV is neither
I would expect to see true (it's numeric) and false for notV

I have no idea what programming language you're using here, so I can't tell if you have syntax errors involved. For instance, are numeric and int the only two possible types for numbers in the language you're using? I'm assuming that "OneOf() is the same as an "or" test. Is that the case? What is the "not" version of "OneOf()"?

**Comment by wvervuurt** [comment]
Thanks for chipping in Max, but it seems like you are unaware that this is an issue report for Validation, the PHP Validation library that is this GitHub project. If you don't know the library and don't even recognize the language I don't think you can help me with this. Thanks for trying though!

Does anyone else encounter this bug or does it ring any bells for the maintainers of this awesome library?

**Comment by henriquemoody** [comment]
I wrote this code:
```php
$noneOfRule         = v::noneOf(v::numeric(), v::int());
$noneOfRuleNagative = v::not($noneOfRule);

$oneOfRule          = v::oneOf(v::numeric(), v::int());
$oneOfRuleNagative  = v::not($oneOfRule);

echo 'noneOfRule:           '.var_export($noneOfRule->validate(13.37), true).PHP_EOL;
echo 'noneOfRuleNagative:   '.var_export($noneOfRuleNagative->validate(13.37), true).PHP_EOL;

echo str_repeat('-', 80).PHP_EOL;

echo 'oneOfRule:            '.var_export($oneOfRule->validate(13.37), true).PHP_EOL;
echo 'oneOfRuleNagative:    '.var_export($oneOfRuleNagative->validate(13.37), true).PHP_EOL;
echo str_repeat('-', 80).PHP_EOL;
```

And I'm getting:

```
noneOfRule:           false
noneOfRuleNagative:   false
--------------------------------------------------------------------------------
oneOfRule:            true
oneOfRuleNagative:    true
--------------------------------------------------------------------------------
```

I was expecting:

```
noneOfRule:           false
noneOfRuleNagative:   true
--------------------------------------------------------------------------------
oneOfRule:            true
oneOfRuleNagative:    false
--------------------------------------------------------------------------------
```

Clearly we have a bug here.

I already figured out what is happening, I will make a fix soon.

**Comment by wvervuurt** [comment]
Great, thanks!

**Comment by henriquemoody** [comment]
This issue was fixed by #332 and it's already available on **0.7.9** and **0.8.5**  versions.

Thanks for reporting!
